lms/static/scripts/frontend_apps/components/test/StudentSelector-test.js

```diff
@@ -0,0 +1,90 @@
+import { createElement } from 'preact';
+import { shallow } from 'enzyme';
+
+import StudentSelector from '../StudentSelector';
+
+describe('StudentSelector', () => {
+  const renderSelector = (props = {}) => {
+    const fakeOnSelectStudent = sinon.stub();
+    const fakeSelectedStudentIndex = 0;
+    const fakeStudents = [
+      {
+        username: 'user1',
+        displayName: 'User 1',
+      },
+      {
+        username: 'user2',
+        displayName: 'User 2',
+      },
+    ];
+
+    return shallow(
+      <StudentSelector
+        onSelectStudent={fakeOnSelectStudent}
+        selectedStudentIndex={fakeSelectedStudentIndex}
+        students={fakeStudents}
+        {...props}
+      />
+    );
+  };
+
+  it('shall have "All Students" as the default option', () => {
+    const wrapper = renderSelector({ selectedStudentIndex: -1 });
+    assert.equal(wrapper.find('select [selected=true]').text(), 'All Students');
+  });
+
+  it('sets the selected option should to the second user', () => {
+    const wrapper = renderSelector({ selectedStudentIndex: 1 });
+    assert.equal(wrapper.find('select [selected=true]').text(), 'User 2');
+  });
+
+  it('calls the onSelectStudent callback when the select list is changed', () => {
```

Comment:
Oh! This test description threw me for a minute. Perhaps more accurate/clear to say something like `'...when the select input value changes'` or something?

Reply:
cleaned it up